com.unity.shadergraph/Editor/ShaderFoundry/Utilities/ShaderGenerator.cs

```diff
@@ -102,5 +103,25 @@ static void GenerateSubShaders(ShaderBuilder builder, ShaderContainer container,
                 linker.Link(builder, container, templateInst);
             }
         }
+
+        static void GenerateDependencies(ShaderBuilder builder, ShaderContainer container, ShaderInstance shaderInst)
+        {
+            foreach (var dependency in shaderInst.Dependencies)
+                builder.AppendLine($"Dependency \"{dependency.DependencyName}\" = \"{dependency.ShaderName}\"");
+        }
+
+        static void GenerateCustomEditors(ShaderBuilder builder, ShaderContainer container, ShaderInstance shaderInst)
+        {
+            foreach (var customEditor in shaderInst.CustomEditors)
+                builder.AppendLine($"CustomEditorForRenderPipeline \"{customEditor.CustomEditorClassName}\" \"{customEditor.RenderPipelineAssetClassName}\"");
```

Comment:
Does this handle the case that I think was still in built-in where it did the old method (non render pipeline specific)?

Reply:
The built-in editor and user-specified override is supported via this path.
But I'm thinking the "ShaderGraph default editor" is not...   you would only get that currently if you inspect a material while using a render pipeline that is not supported by the graph.   But we might want to add support for injecting the ShaderGraph default to keep it equivalent.